server/controllers/memoCtrl.js

```diff
@@ -533,6 +540,23 @@ async function getTermsWithCourseRounds(req, res, next) {
   }
 }
 
+async function _getAllRoundsWithApplicationCodes(courseCode, responseLanguage) {
+  //let allRounds = await getCourseRoundTerms(courseCode)
+  let allRounds = await getDetailedInformation(courseCode, responseLanguage)
+  let allTempRounds = []
+  const rounds = []
+  allRounds.roundInfos.map(t => {
+    if (isDateWithInCurrentOrFutureSemester(t.round.firstTuitionDate, t.round.lastTuitionDate)) {
```

Comment:
All these variables `t.round.something` should be safe extracted before using them, it often happens some of these values are missing, because `t.round` can be missing

Reply:
coden is refactored